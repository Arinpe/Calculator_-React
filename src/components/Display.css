.display {
  display: flex;
  align-items: center;
  background-color: #5c5c8a;
  width: 700px;
  height: 100px;
  justify-content: flex-end;
}

.result {
  color: #fff;
  font-weight: 2;
  padding-right: 20px;
}

.seperate {
  display: flex;
  justify-content: space-between;
}

.el {
  font-size: 25px;
  padding-top: 40px;
}
