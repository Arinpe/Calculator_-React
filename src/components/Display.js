import PropTypes from 'prop-types';

export default function Display({ memor, current }) {
  return (
    <div className="display">
      <div className="displayupper">{memor}</div>
      <div className="displaylower">{current}</div>
    </div>
  );
}

Display.defaultProps = {
  memor: '',
  current: '0',
};

Display.propTypes = {
  memor: PropTypes.string,
  current: PropTypes.string,
};
