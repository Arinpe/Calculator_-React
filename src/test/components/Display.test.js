import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../../components/Display';

describe('Display', () => {
  let page;
  const memor = 'Test Memor';
  const current = 'Test Current';

  const setUp = () => {
    page = render(<Display memor={memor} current={current} />);
  };

  test('should render', () => {
    setUp();
    const displayElement = page.container.querySelector('.display');
    expect(displayElement).toBeTruthy();
  });

  test('should render', () => {
    setUp();
    const displayElement = page.container.querySelector('.display');
    expect(displayElement).not.toBeFalsy();
  });

  test('should render memor prop', () => {
    setUp();
    const memorElement = page.container.querySelector('.displayupper');
    expect(memorElement.textContent).toBe(memor);
  });

  test('memor prop should not be blank', () => {
    setUp();
    const memorElement = page.container.querySelector('.displayupper');
    expect(memorElement.textContent).not.toBeFalsy();
  });

  test('should render current prop', () => {
    setUp();
    const currentElement = page.container.querySelector('.displaylower');
    expect(currentElement.textContent).toBe(current);
  });

  test('current prop should not be blank', () => {
    setUp();
    const currentElement = page.container.querySelector('.displaylower');
    expect(currentElement.textContent).not.toBeFalsy();
  });
});
