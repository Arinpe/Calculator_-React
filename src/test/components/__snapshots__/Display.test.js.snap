// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render without crashing 1`] = `
<div
  className="display"
>
  <div
    className="displayupper"
  >
    
  </div>
  <div
    className="displaylower"
  >
    0
  </div>
</div>
`;
